#! /bin/sh /usr/share/dpatch/dpatch-run
## ppd-poll-with-client-conf.dpatch by  <till.kamppeter@gmail.com>
##
## All lines beginning with `## DP:' are a description of the patch.
## DP: If an external server is used via client.conf and the DNS is
## DP: inconsistent (ex: DNS gives "noname" for many IPs, reverse DNS
## DP: gives one of these IPs for "noname") local PPDs can get polled
## DP: for print queues instead of the PPDs of the external server
## DP: (http://www.cups.org/str.php?L2763)

@DPATCH@
diff -urNad cups-1.4.0~svn8773~/cups/util.c cups-1.4.0~svn8773/cups/util.c
--- cups-1.4.0~svn8773~/cups/util.c	2009-08-23 12:16:16.000000000 +0200
+++ cups-1.4.0~svn8773/cups/util.c	2009-08-23 17:42:54.000000000 +0200
@@ -1083,11 +1083,13 @@
     http2 = http;
   else if ((http2 = httpConnectEncrypt(hostname, port,
                                        cupsEncryption())) == NULL)
-  {
-    DEBUG_puts("1cupsGetPPD3: Unable to connect to server!");
+    if ((http2 = httpConnectEncrypt(http_hostname, http_port,
+				    cupsEncryption())) == NULL)
+    {
+      DEBUG_puts("1cupsGetPPD3: Unable to connect to server!");
 
-    return (HTTP_SERVICE_UNAVAILABLE);
-  }
+      return (HTTP_SERVICE_UNAVAILABLE);
+    }
 
  /*
   * Get a temp file...
